### PR TITLE
SNOW-1373790: Clean up pandas docstrings + supported methods

### PR DESCRIPTION
**Pull request by sfc-gh-joshi**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1373790

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

This PR prunes some duplicated + unsupported methods from Snowpark pandas API docs, and adds more comprehensive docstrings (mostly copied from pandas) for other methods. This PR does not add any doctests, nor does it change any meaningful code.

All changes are listed below:
_De-duplicated listings_:
- Series
  - is_unique
  - duplicated
- DataFrame
  - round
Note that Series/DF head and tail are deliberately left duplicated; this matches pandas documentation, as they are mentioned under both the "Indexing, iteration" and "Reindexing / selection" headings.

_Removed unimplemented listing_:
- Series
  - kurtosis
- SeriesGroupBy
  - apply
- Resampler
  - groups
  - indices
  - get_group
  - apply
  - aggregate
  - transform
  - bfill
  - nearest
  - fillna
  - asfreq
  - nunique
  - first
  - last
  - interpolate
  - ohlc
  - pad
  - pipe
  - prod
  - quantile
  - sem
  - size
- Rolling
  - aggregate
  - apply
  - corr
  - count
  - cov
  - kurt
  - median
  - quantile
  - rank
  - sem
  - skew

_Added implemented listing_:
I did not comprehensively look for implemented methods that were not listed, these were just a few methods that I noticed in the course of checking other APIs.
- SeriesGroupBy
  - head
  - idxmax
  - idxmin
  - nunique
  - tail

_Improved documentation_:
- pd
  - qcut (I'm not sure why it wasn't inheriting pandas docs, but we should override them anyway since we don't implement all parameters)
- BasePandasDataset
  - convert_dtypes
  - rename_axis
  - values
  - ffill/pad
- Series
  - name
  - empty
  - hasnans
  - ndim
  - shape
  - rename_axis
  - quantile
- DataFrame
  - empty
  - quantile
  - select_dtypes
- GroupBy
  - std
  - var
  - rank
  - nunique
  - quantile
  - __iter__

### CODE REVIEW

File: docs/source/modin/dataframe.rst [hunk 1]
```diff
@@ -77,7 +77,6 @@ DataFrame
     DataFrame.rfloordiv
     DataFrame.rmod
     DataFrame.rpow
-    DataFrame.round
```
Comment (sfc-gh-azhan) [inline docs/source/modin/dataframe.rst:80]: we implemented round already cc @sfc-gh-helmeleegy

Comment (sfc-gh-joshi) [reply docs/source/modin/dataframe.rst:80]: It's still in the doc, this was just a duplicate entry.

File: docs/source/modin/supported/dataframe_supported.rst [hunk 2]
```diff
@@ -65,14 +65,15 @@ Methods
...
-|                             |                                 |                                  | the default value or a scalar.                     |
-|                             |                                 |                                  | if ``axis == 1``: ``Y`` when function is `count`,  |
-|                             |                                 |                                  | `min`, `max`, or `sum` and the index is not a      |
-|                             |                                 |                                  | MultiIndex.                                        |
+| ``agg``                     | P                               | ``margins``, ``observed``,       | If ``axis == 0``: ``Y`` when  function is one of   |
+|                             |                                 | ``sort``                         | ``count``, ``mean``, ``min``, ``max``, ``sum``,    |
+|                             |                                 |                                  |``median``; ``std`` and ``var`` supported with      |
```
Comment (sfc-gh-vbudati) [inline docs/source/modin/supported/dataframe_supported.rst:70]:
```suggestion
|                             |                                 |                                  | ``median``; ``std`` and ``var`` supported with     |
```

File: docs/source/modin/supported/series_supported.rst [hunk 3]
```diff
@@ -76,11 +78,11 @@ Methods
...
-|                             |                                 |                                  | `min`, `max`, `sum`, `median`; `std` and `var`     |
-|                             |                                 |                                  | supported with `ddof=0` or `ddof=1`; `quantile` is |
-|                             |                                 |                                  | supported when `q` is the default value or a       |
-|                             |                                 |                                  | scalar                                             |
+| ``agg``                     | P                               |                                  | ``Y`` when  function is one of ``count``,          |
+|                             |                                 |                                  | ``mean``, ``min``, ``max``, ``sum``, ``median``;   |
+|                             |                                 |                                  | ``std`` and ``var`` supported with ``ddof=0`` or   |
```
Comment (sfc-gh-vbudati) [inline docs/source/modin/supported/series_supported.rst:83]:
```suggestion
|                             |                                 |                                  | ``std``, or ``var`` supported with ``ddof=0`` or   |
```

File: docs/source/modin/supported/series_supported.rst [hunk 4]
```diff
@@ -221,10 +223,6 @@ Methods
 +-----------------------------+---------------------------------+----------------------------------+----------------------------------------------------+
 | ``hist``                    | N                               |                                  |                                                    |
 +-----------------------------+---------------------------------+----------------------------------+----------------------------------------------------+
-| ``idxmax``                  | N                               |                                  |                                                    |
```
Comment (sfc-gh-vbudati) [inline docs/source/modin/supported/series_supported.rst:224]: I think `idxmax` and `idxmin` should also work for Series. We also have tests for it.
https://pandas.pydata.org/docs/reference/api/pandas.Series.idxmax.html

File: docs/source/modin/supported/series_supported.rst (hunk 3 above)
Comment (sfc-gh-joshi) [reply docs/source/modin/supported/series_supported.rst]: This isn't a continuation of the list, ``std`` and ``var`` are only supported with ``ddof=0`` (that's why they're separated from the other aggregations by the semicolon).

File: docs/source/modin/supported/series_supported.rst (hunk 4 above)
Comment (sfc-gh-joshi) [reply docs/source/modin/supported/series_supported.rst:224]: They are, these were just duplicated. I just realized they're not alphabetized properly though, so I've moved `hist` to the proper location.

**Review by sfc-gh-vbudati — APPROVED** [review]
Thanks Jonathan!!

**Review by sfc-gh-helmeleegy — APPROVED** [review]
LGTM!

**Pull request merged**